src/Relation/Binary/Construct/Closure/SymmetricTransitive.agda

```diff
@@ -0,0 +1,121 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Symmetric transitive closures of binary relations
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Relation.Binary.Construct.Closure.SymmetricTransitive where
+
+open import Level
+open import Function
+open import Relation.Binary
+
+private
+  variable
+    a ℓ ℓ′ : Level
+    A B    : Set a
+
+module _ {A : Set a} (_≤_ : Rel A ℓ) where
+  private
+    variable
+      x y z : A
+
+  data Plus⇔ : Rel A (a ⊔ ℓ) where
+    forth  : x ≤ y → Plus⇔ x y
+    back   : y ≤ x → Plus⇔ x y
+    forth⁺ : x ≤ y → Plus⇔ y z → Plus⇔ x z
+    back⁺  : y ≤ x → Plus⇔ y z → Plus⇔ x z
+
+module _ (_∼_ : Rel A ℓ) where
+
+  trans : Transitive (Plus⇔ _∼_)
+  trans (forth r) rel′      = forth⁺ r rel′
+  trans (back r) rel′       = back⁺ r rel′
+  trans (forth⁺ r rel) rel′ = forth⁺ r (trans rel rel′)
+  trans (back⁺ r rel) rel′  = back⁺ r (trans rel rel′)
+
+  sym : Symmetric (Plus⇔ _∼_)
+  sym (forth r)      = back r
+  sym (back r)       = forth r
+  sym (forth⁺ r rel) = trans (sym rel) (back r)
+  sym (back⁺ r rel)  = trans (sym rel) (forth r)
+
+  isPartialEquivalence : IsPartialEquivalence (Plus⇔ _∼_)
+  isPartialEquivalence = record
+    { sym   = sym
+    ; trans = trans
+    }
+
+  partialSetoid : PartialSetoid _ _
+  partialSetoid = record
+    { Carrier              = A
+    ; _≈_                  = Plus⇔ _∼_
+    ; isPartialEquivalence = isPartialEquivalence
+    }
+
+  module _ (refl : Reflexive _∼_) where
+
+    isEquivalence : IsEquivalence (Plus⇔ _∼_)
+    isEquivalence = record
+      { refl  = forth refl
+      ; sym   = sym
+      ; trans = trans
+      }
+
+    setoid : Setoid _ _
+    setoid = record
+      { Carrier       = A
+      ; _≈_           = Plus⇔ _∼_
+      ; isEquivalence = isEquivalence
+      }
+
+  module _ {c e} (S : Setoid c e) where
+    private
+      module S = Setoid S
+
+    minimal : (f : A → Setoid.Carrier S) →
+              _∼_ =[ f ]⇒ Setoid._≈_ S →
+              Plus⇔ _∼_ =[ f ]⇒ Setoid._≈_ S
```

Comment:
What's the intuition behind the name `minimal`?

Reply:
if `f` converts `_∼_` to an equivalence, it can also convert the symmetric transitive closure to the equivalence. `minimal` might not be the best name for this function.